Sources/Starknet/Network/StarknetRequest.swift

```diff
@@ -140,6 +140,12 @@ public enum RequestBuilder {
         return StarknetRequest(method: .getEvents, params: .getEvents(params))
     }
 
+    public static func getStorageProof(blockId: StarknetBlockId, classHashes: [Felt]?, contractAddresses: [Felt], contractStorageKeys: [ContractStorageKey]) -> StarknetRequest<StarknetGetStorageProofResponse> {
+        let params = GetStorageProofParams(blockId: blockId, classHashes: classHashes, contractAddresses: contractAddresses, contractStorageKeys: contractStorageKeys)
+
+        return StarknetRequest(method: .getStorageProof, params: .getStorageProof(params))
+    }
```

Comment:
Why are `classHashes` optional, but `contractAddresses` and `contractStorageKeys` aren't? Aren't they all optional per the spec? That is, unless empty array is equivalent to nil in this case, but than that'd be something to be addressed on the spec side 🤔

Reply:
Indeed they should be optional, I added it to `GetStorageProofParams`, just missed it in `getStorageProof`.